builder/evaluator.go

```diff
@@ -191,6 +197,18 @@ func (b *builder) Run(context io.Reader) (string, error) {
 		}
 	}
 
+	// check if there are any leftover build-args that were passed but not
+	// consumed during build. Return an error, if there are any.
+	leftoverArgs := []string{}
```

Comment:
Yes we can do it that way.

I just latched this check at end as by then we have the parsed state of `b.Config.AllowedBuildArgs` ready and I was able to depend on ARG evaluator and dispatcher function to throw out any errors etc.

I think if I have to do it before real dispatch I can walk over `b.dockerfile.Children` and look at the parsed children for `ARG` nodes. Let me try it out in a separate commit in this PR and see how it comes out.